src/modules/esc_battery/EscBattery.cpp

```diff
@@ -0,0 +1,174 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "EscBattery.hpp"
+#include <drivers/drv_hrt.h>
+
+using namespace time_literals;
+
+EscBattery::EscBattery() :
+	ModuleParams(nullptr),
+	WorkItem(MODULE_NAME, px4::wq_configurations::lp_default),
+	_battery(1, this)
+{
+	parameters_updated();
+}
+
+EscBattery::~EscBattery()
+{
+}
+
+bool
+EscBattery::init()
+{
+	if (!_esc_status_sub.registerCallback()) {
+		PX4_ERR("esc_status callback registration failed!");
+		return false;
+	}
+
+	return true;
+}
+
+void
+EscBattery::parameters_updated()
+{
+	ModuleParams::updateParams();
+}
+
+void
+EscBattery::Run()
+{
+	if (should_exit()) {
+		_esc_status_sub.unregisterCallback();
+		exit_and_cleanup();
+		return;
+	}
+
+	if (_parameter_update_sub.updated()) {
+		// Clear update
+		parameter_update_s param_update;
+		_parameter_update_sub.copy(&param_update);
+
+		parameters_updated();
+	}
+
+	esc_status_s esc_status;
+
+	if (_esc_status_sub.update(&esc_status)) {
+
+		if (hrt_elapsed_time(&esc_status.timestamp) > 500_ms ||
+		    esc_status.esc_count == 0 ||
+		    esc_status.esc_count > 8) {
```

Comment:
Why hardcode to 8?

Reply:
I'm changing it to the define from the msg file.